### PR TITLE
regression test for gap_fit

### PR DESCRIPTION
**Pull request by jameskermode**

### CODE REVIEW

**Comment by jameskermode** [comment]
@albapa an initial attempt at a regression test based on the cut down Si training set. On my Mac at least I don't get the same alpha to high precision on repeated runs, even with fixed random seed and fixed sparse point set. Would you be able to have a look if I'm doing something wrong?

**Comment by jameskermode** [comment]
Can run this individual test from command line with

```shell
cd QUIP/tests
HAVE_GAP=1 python3 test_gapfit.py
```

**Comment by jameskermode** [comment]
The `IEEE_DENORMAL` crash shows up here too, so may well be a problem with BLAS libraries used in the CI/pip builds

https://github.com/libAtoms/QUIP/pull/308/checks?check_run_id=2853325368

**Comment by jameskermode** [comment]
Think I've fixed it - `HAVE_QR` was not set to 1 in the Makefile.inc used for CI.

**Comment by albapa** [comment]
Looks great! Which merge strategy is preferred?

**Comment by jameskermode** [comment]
Just cleaning it up a tiny bit, then I can merge it directly into public.

I removed the RMSE comparisons with training data and left only those with the reference GAP, do you agree with that? Since this is verification, not validation.

**Comment by jameskermode** [comment]
Could probably do with being a bit faster too - shall I cut to 500 sparse points and halve the training set, or is that too small to be representative?

**Comment by albapa** [comment]
I agree with the removal of RMSEs - all we care about here are the weights.

For this test, I'd expect a very basic training and sparse set would work as the weights are quite a sensitive hash...

**Comment by jameskermode** [comment]
Passing with a max absolute error on the alpha of 1e-2. Good enough?

I've generalised a little bit so it will be easy to add further tests - copy the silicon example, run once with `new_test=True` to generate a JSON file, then change to `new_test=False` and commit the `test_gapfit.py` and new JSON.

**Comment by jameskermode** [comment]
I'm happy for this to be merged when you are

**Comment by albapa** [comment]
I think this is great and ready to be merged.
